### PR TITLE
Add better 404 page

### PR DESCRIPTION
**Pull request by dawran6**
This PR addresses issue #32.

1. Add custom interceptor for handling 404 error
1. Add ns, `cljdoc.render.error`, for rendering the error page

Here's a snapshot of the 404 page:

screenshot_2018-10-18 screenshot

On smaller mobile device (iPhone 6)
screenshot_2018-10-18 screenshot 1

On larger mobile device (iPhone 6 plus)
screenshot_2018-10-18 screenshot 2

Any feedback is welcome :) Thanks!

### CODE REVIEW

**Comment by avichalp** [comment]
This looks good to me 🚀 
Thanks @dawran6 😄

File: src/cljdoc/server/pedestal.clj [hunk 1]
```diff
@@ -263,6 +264,16 @@
            (fn [request opts]
              (etag/add-file-etag request false)))})
 
+(def not-found-interceptor
+  {:name ::not-found-interceptor
+   :leave (fn [context]
+            (if-not (http/response? (:response context))
+              (do (log/info ::not-found-interceptor)
```
[the remark this answers is not in the document]
Comment (dawran6) [reply src/cljdoc/server/pedestal.clj]: I wasn’t sure if logging such information would be useful for production. After a second thought, spamming the logging with every single not-found request might not be a good idea. Thanks for catching that.

**Comment by dawran6** [comment]
Thanks y’all for the comments! I’d like to address them in this same PR.

**Comment by martinklepsch** [comment]
@dawran6 looks great, merged & [deployed](https://cljdoc.xyz/not-found)! 🎉 

One thing we could still add is a link to the home page. Maybe just the logo on top of the message with a link...

I'll invite you to the cljdoc team on GitHub which gives you write access to this repo. Feel free to push this directly. I'm still happy to review the commit if you'd like me to. 🙂

**Pull request merged**